src/main/java/org/broadinstitute/hellbender/engine/ContextShard.java

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.collections.IntervalsSkipListOneContig;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * Immutable storage class.
+ * This class enforces no consistency guarantees. The intent of course
+ * is for the reads and variants to be in the interval, but that's up the callers.
```

Comment:
well I meant the variants in ReadContext. In order to make sure that those are all there, we put in a few extras in variants temporarily (we empty the "variants" field completely once we have the ReadContext). Your second question is ambiguous because I don't know what you mean by "records". 

For BQSR the key requirement is that every read is present in at most one shard (zero if it's unaligned and has no aligned mate). I believe that's the case now - if not then that would be a bug.

I've changed the comment.